src/ui/authenticationInApp/ViewAuthenticationSignIn.qml

```diff
@@ -10,31 +10,20 @@ import components 0.1
 import components.inAppAuth 0.1
 
 VPNInAppAuthenticationBase {
-
-    // TODO
-    // We are completing an authentication using an existing account.
-    // There is nothing to do here, except... waiting.
-    // There 5 possible next-steps:
-    // - authentication completed (VPN.state will change, the authentication is
-    //   completed)
-    // - unblock code needed. This can happen for security reasons. We go
-    //   to UnblockCodeNeeded. The user needs to insert the 6-digit code.
-    // - The user enters the wrong password, sees error.
-    // - The user clicks "Change email" or the back arrow and goes back to start.
-    // - Some other error, goes back to start and sees error.
-
-    // TODOs (likely there are more)
-    // Open forgot password flow in webview on click
-    // Form interaction polish
-    // Add password criteria tooltip
-    // Maybe add button loader
-
-    _changeEmailLinkVisible: true
+    _changeEmailLinkVisible: !isDeleteAccountAuth
     _menuButtonImageSource: "qrc:/nebula/resources/back.svg"
-    _menuButtonOnClick: () => { VPNAuthInApp.reset() }
+    _menuButtonOnClick: () => {
+        if (isDeleteAccountAuth) {
+            cancelAccountDeletion();
+        } else {
+            VPNAuthInApp.reset();
+        }
+    }
     _menuButtonAccessibleName: qsTrId("vpn.main.back")
     _headlineText: VPNAuthInApp.emailAddress
-    _subtitleText: VPNl18n.InAppAuthSignInSubtitle
+    _subtitleText: !isDeleteAccountAuth
```

Comment:
We're doing the `isDeleteAccountAuth` quite a few times in this file and I wonder if it might feel more manageable to do it all in one place with something like:

```
states:[
  State {
    when: isDeleteAccountAuth
    PropertyChanges {...}

Reply:
Good suggestion — done!